sdk/search/azure-search/src/main/java/com/azure/search/models/WebApiSkill.java

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// Code generated by Microsoft (R) AutoRest Code Generator.
+// Changes may cause incorrect behavior and will be lost if the code is
+// regenerated.
+
+package com.azure.search.models;
+
+import com.azure.core.annotation.Fluent;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.time.Duration;
+import java.util.Map;
+
+/**
+ * A skill that can call a Web API endpoint, allowing you to extend a skillset
+ * by having it call your custom code.
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type")
+@JsonTypeName("#Microsoft.Skills.Custom.WebApiSkill")
+@Fluent
+public final class WebApiSkill extends Skill {
+    /*
+     * The url for the Web API.
+     */
+    @JsonProperty(value = "uri", required = true)
+    private String uri;
+
+    /*
+     * The headers required to make the http request.
+     */
+    @JsonProperty(value = "httpHeaders")
+    private Map<String, String> httpHeaders;
```

Comment:
This should technically be Map<String, Object> to be synonymous with the backend

Reply:
The backend implementation of models does not dictate our API surface. I bet it's `object` in the backend in name only -- I'd be very surprised if we allow values other than strings in that context.

Reply:
You're right, we do validation to only let you put PUT if the value is a string.  I wonder why we implemented it as <string, object> in the first place then...

Reference: AzureSearch\Source\Search\Product\SearchCore\Models\VersionNeutral\Enrichment\WebApiHttpHeaders.cs

Reply:
@Careyjmac It could be some OData esoterica. It doesn't cope particularly well with dictionaries.

We'll consider this resolved then.